spec/features/users_feature_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+feature "User can sign in and out" do
+  context "user not signed in and on the homepage" do
+    it "should see a 'sign in' link and a 'sign up' link" do
+      visit('/')
+      expect(page).to have_link('Sign in')
+      expect(page).to have_link('Sign up')
+    end
+
+    it "should not see 'sign out' link" do
+      visit('/')
+      expect(page).not_to have_link('Sign out')
+    end
+  end
+
+  context "user signed in on the homepage" do
+    before do
+      sign_up
+    end
+
+    it "should see 'sign out' link" do
+      visit('/')
+      expect(page).to have_link('Sign out')
+    end
+
+    it "should not see a 'sign in' link and a 'sign up' link" do
+      visit('/')
+      expect(page).not_to have_link('Sign in')
+      expect(page).not_to have_link('Sign up')
+    end
+  end
+
+  feature "When users are logged in" do
+
+  it "they can only edit their own pictures" do
+    sign_up
+    upload_picture
+    click_link 'Sign out'
+    sign_up_2
+    click_link 'Edit PFC'
+    expect(current_path).to eq '/pictures'
+    expect(page).to have_content 'Only the creator can edit this picture'
+  end
+
+  it "they can only delete their own pictures" do
+    sign_up
+    upload_picture
+    click_link 'Sign out'
+    sign_up_2
+    expect(current_path).to eq '/'
+    click_link 'Delete PFC'
+    expect(current_path).to eq '/pictures'
+    expect(page).to have_content 'Only the creator can delete this picture'
+  end
+end
```

Comment:
`end` at 56, 0 is not aligned with `feature "When users are logged in" do` at 34, 2